lib/SPIRVOp.cpp

```diff
@@ -65,7 +68,14 @@ Instruction *InsertSPIRVOp(Instruction *Insert, spv::Op Opcode,
     ArgValues.push_back(Arg);
   }
 
-  return CallInst::Create(func, ArgValues, "", Insert);
+  Instruction *NewInst = CallInst::Create(func, ArgValues, "", Insert);
+
+  // Set the source location for the new instruction
+  if (DbgLoc) {
+    NewInst->setDebugLoc(DbgLoc);
+  }
```

Comment:
```suggestion
  NewInst->setDebugLoc(Insert->getDebugLoc());
```
I don't think there is a reason to make it conditional.